vm/io.c

```diff
@@ -12,8 +13,34 @@ OBJ TrIO_close(VM, OBJ self) {
   return TR_NIL;
 }
 
+OBJ TrIO_read(VM, OBJ self, OBJ path_arg) {
+  UNUSED(vm);
+  UNUSED(self);
+  char *path = TR_STR_PTR(path_arg);
+  FILE *fp;
+  struct stat stats;
+
+  if (stat(path, &stats) == -1) tr_raise_errno(path);
+  fp = fopen(path, "rb");
+  if (!fp) tr_raise_errno(path);
+
+  char *string = TR_ALLOC_N(char, stats.st_size + 1);
+  if (fread(string, 1, stats.st_size, fp) == (size_t)stats.st_size) {
+    fclose(fp);
+    TrString *s = TR_INIT_CORE_OBJECT(String);
+    s->len = stats.st_size;
+    s->interned = 0;
+    s->ptr = string;
+    s->ptr[s->len] = '\0';
+    return (OBJ)s;
+  }
+  fclose(fp);
```

Comment:
as a function call requires some overhead (putting args into the right regs, adjusting stack, etc) i think it'd be cheaper in terms of binarysize if you have a single fclose() location and jump there with a goto

Reply:
actually you dont even need a goto if you just assign a return value

Reply:
this is fixed in my latest commit, please check